core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala

```diff
@@ -93,16 +95,23 @@ abstract class HistoryServerSuite extends SparkFunSuite with BeforeAndAfter with
       .set(EXECUTOR_PROCESS_TREE_METRICS_ENABLED, true)
       .set(HYBRID_STORE_DISK_BACKEND, diskBackend.toString)
     conf.setAll(extraConf)
-    provider = new FsHistoryProvider(conf)
-    provider.checkForLogs()
+    val usedProvider = appHistoryProvider.getOrElse {
```

Comment:
Is this a related change?

Reply:
Yes, to test whether history server return correct response, the UT added in this file requires to create history server service with a private implementation of provider interface, to throw customized error. But current version only creates `FsHistoryProvider`.

So change here is to allow to create pass private provider to initialization method. The previous init way is still kept, so it won't affect other existing tests.

![image](https://github.com/apache/spark/assets/10705175/c8d220f0-523a-4dda-8f1d-523a8660263a)

![image](https://github.com/apache/spark/assets/10705175/1ed4107b-8f41-49c8-988e-0167da435223)